backends-common/elasticsearch/src/test/java/org/apache/james/backends/es/ElasticSearchHealthCheckConnectionTest.java

```diff
@@ -0,0 +1,69 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.backends.es;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.james.util.docker.Images;
+import org.apache.james.util.docker.RateLimiters;
+import org.apache.james.util.docker.SwarmGenericContainer;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.Client;
+import org.junit.Rule;
+import org.junit.Test;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ElasticSearchHealthCheckConnectionTest {
+
+    private static final int ES_APPLICATIVE_PORT = 9300;
+    private static final Set<IndexName> indices = new HashSet<>();
+
+    private static final WaitStrategy WAIT_STRATEGY = Wait.forHttp("/").forPort(ES_APPLICATIVE_PORT).withRateLimiter(RateLimiters.TWENTIES_PER_SECOND);
```

Comment:
Hi @jreijn, regarding to the fail test we got with CI
```
[bf575b21dcd16d5bb6f5b6114086582584f8d6d1] [ERROR] Errors: 
[bf575b21dcd16d5bb6f5b6114086582584f8d6d1] [ERROR]   ElasticSearchHealthCheckConnectionTest.testHealthCheckValidation » ContainerLaunch
```
> Yes I will take a look at the problem. It's hard to figure out what's wrong though since I can't test the system on my machine. @blackheaven @chibenwa how do you test this on your local machine without dependening on the build system? If it's easier to do on Linux I will just do my work in a VM.

I'm using MacOS and this test is still being fail on my machine. You can setup your Test like the way `ESReporterTest` does. And it can work both on MacOS or Linux machine with that setup. 
```java
    @Rule
    public SwarmGenericContainer esContainer = new SwarmGenericContainer(Images.ELASTICSEARCH)
        .withAffinityToContainer()
        .withExposedPorts(ES_HTTP_PORT, ES_APPLICATIVE_PORT)
        .waitingFor(new HostPortWaitStrategy().withRateLimiter(RateLimiters.TWENTIES_PER_SECOND));

@Test
public void myTest() {
ClientProviderImpl clientProvider = ClientProviderImpl.forHost(esContainer.getHostIp(), esContainer.getMappedPort(ES_APPLICATIVE_PORT));
}
```
you get your ClientProviderImpl by using `forHost(esContainer.getHostIp() // this should be localhost, esContainer.getMappedPort(ES_APPLICATIVE_PORT) // this should be mapped port of 9300 exposed on local machine)`

By doing that, both tcp connection to Docker ES on MacOS and Linux can work. 

And. The reason, for test failure on CI came from http wait strategy, Elasticsearch open 9200 HTTP port, not, 9300 for http. But for java `Client` usages.